app/models/events/pull_request.rb

```diff
@@ -58,8 +58,12 @@ class PullRequest < ApplicationRecord
               inclusion: { in: [true, false] }
     validates :github_id, uniqueness: true
 
-    ##
-    # Returns the number of reviews this pull_request has
-    delegate :count, to: :reviews, prefix: true
+    after_validation :build_merge_time, on: :update, if: :merged_at_changed?
```

Comment:
Perhaps the following is a bit simpler:

`after_update :build_merge_time, if: :merged_at_changed?`

WDYT?

Reply:
After_update is obsolete https://apidock.com/rails/v2.3.8/ActiveRecord/Callbacks/after_update.